db/compaction_picker.cc

```diff
@@ -1275,6 +1294,41 @@ bool LevelCompactionPicker::PickCompactionBySize(VersionStorageInfo* vstorage,
   return inputs->size() > 0;
 }
 
+bool LevelCompactionPicker::PickIntraL0Compaction(
+    VersionStorageInfo* vstorage, CompactionInputFiles* inputs) {
+  inputs->clear();
+  const std::vector<FileMetaData*>& level_files =
+      vstorage->LevelFiles(0 /* level */);
+  if (level_files.size() == 0 || level_files[0]->being_compacted) {
```

Comment:
Let me try it. The latest benchmark showed a new problem: after compacting large L0 files to base level, the base level's compaction score can become very high (~4). Then L0 compaction won't be scheduled for a long time since its compaction score can't really grow beyond 2.5 (write delay trigger / compaction trigger, so 10 / 4 for MyRocks). Couple ideas:

(1) we can change compaction score to grow exponentially higher in L0 when approaching write delay trigger
(2) we can cap the L0 file size like my initial implementation did. But it's hard to pick a good size and we don't want to add options.